src/App.tsx

```diff
@@ -1,12 +1,25 @@
-import React from 'react';
-import TrafficLight from './Components/TrafficLight';
-import './App.css';
+import React, { useEffect, useState } from "react";
+import TrafficLight from "./Components/TrafficLight";
+import "./App.css";
+
+const lightDurations = [1000, 2000, 4000];
 
 function App() {
+  const [activeColor, setActiveColor] = useState<number>(0);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setActiveColor((activeColor + 1) % 3);
```

Comment:
You're creating a new timer whenever activeColor is changed. If you want to read the state without adding it to dependency, you can do like
`setState(prevState => prevState + 1)`